src/tar.rs

```diff
@@ -269,13 +277,17 @@ pub fn create<W: Write>(
                     &mut visited_paths,
                     entry.path(),
                     strip_root(
-                        &destination_dir.join(entry.path().strip_prefix(&source_dir).map_err(
-                            failure::system(format!(
-                                "Unable to relativize path {} with respect to {}.",
-                                entry.path().to_string_lossy().code_str(),
-                                source_dir.to_string_lossy().code_str(),
-                            )),
-                        )?),
+                        &destination_dir.join(
+                            entry
+                                .path()
+                                .strip_prefix(&source_dir)
+                                .map_err(failure::system(format!(
+                                    "Unable to relativize path {} with respect to {}.",
+                                    entry.path().to_string_lossy().code_str(),
+                                    source_dir.to_string_lossy().code_str(),
+                                )))
+                                .map(|p| strip_relative(p))?,
```

Comment:
Hi @luogni, it seems like for paths that go "up" (e.g., `../foo`), this code is just going to just put them in the task's `location`? In other words, `foo` and `../foo` would collide in the container. Am I understanding the behavior correctly?

Reply:
Oh, I think my understanding was slightly wrong. `../foo` on the host would still go to `<location>/../foo` in the container, unless that would result in going higher than the filesystem root, in which case it would just end up at `/foo` in the container. Is that right?

Reply:
yes, only initial ../ will be removed.
*  ../foo will go to /foo
*  ../../blah will go to /blah
*  ./foo/blah/../ will go to /foo/

../foo and ./foo will collide the same way as ./foo and ./foo/blah/../ so I don't think there is any new problem here

Reply:
sorry I didn't get the question, ../ will be removed before appending to <location> so:
*  ../foo will go to <location>/foo

No files will go outside of <location>.

Reply:
> ../foo and ./foo will collide the same way as ./foo and ./foo/blah/../ so I don't think there is any new problem here

But in the `./foo` + `./foo/blah/../` case, those two paths refer to the same file/directory right? I'm more worried about having two distinct files on the host map to the same location in the container.

> No files will go outside of \<location\>

I might be misunderstanding. But consider this situation:

- `location` = `/scratch`
- `input_paths` = `['../foo']`

Then Toast will copy the file to `/scratch/../foo`, which is the same as `/foo`. That's outside of `location`, right?

Reply:
If I didn't do anything wrong it'll copy the file to /scratch/foo because the "../" will be removed by strip_relative before joining input_path to location.

Reply:
Ah, I think I misread the code. But what about this case:

- `location` = `/scratch`
- `input_paths` = `['../foo', 'foo']`

Then those two files will collide in the container filesystem, right?

Without this change, there is essentially a 1:1 mapping between input paths on the host and the files and folders copied into the container. But after this change, it's not clear which file will end up in the container when `input_paths` = `['../foo', 'foo']`. Unfortunately, I think that's a blocker.